docs/internal/GeneralArchitectureGuide.md

```diff
@@ -101,6 +101,24 @@ are coordinated.
 > does not hold, in those cases you can locate the transport action for a REST action by looking at the `NodeClient` invocation in the
 > `Rest*Action`'s `prepareRequest` implementation, it should specify the `ActionType` being invoked which can then be used to locate
 > the `Transport*Action` class that handles it.
+>
+> A netty [EventLoop] thread handles the initial steps of a Rest*Action request lifecycle such as decoding, validation and routing.
+> Upon entry into the "transport layer", [NodeClient] delegates the decision of execution to individual [TransportAction]. Each action
+> determines whether to execute synchronously on invoking thread-an approach reserved for lightweight processing-or to dispatch execution to an appropriate
+> thread pool, which is recommended practice for heavy workloads. Comprehensive mechanisms are available for propagating [ThreadContext] when tasks
+> are dispatched to alternate thread pools.
+>
+> [TransportAction] can also be initiated through peer-to-peer communication between nodes. In such cases, the [InboundHandler]
+> locates the appropriate [TransportAction] by consulting the [NamedRegistry], then invokes its handleExecution() method. When a [TransportAction]
```

Comment:
```suggestion
> locates the appropriate [TransportAction] by consulting the [NamedRegistry], then invokes the [TransportAction]'s handleExecution() method. When a [TransportAction]
```
"its" seems to refer to InboundHandler, but I think you mean TransportAction, right?